pkg/storage/swift/swift.go

```diff
@@ -71,7 +71,23 @@ func GetConfig(listers *regopclient.Listers) (*Swift, error) {
 				return nil, fmt.Errorf("failed to unmarshal clouds credentials: %v", err)
 			}
 
-			if cloud, ok := clouds.Clouds["openstack"]; ok {
+			var cloudName string
+			cloudInfra, err := listers.Infrastructures.Get("cluster")
+			if err != nil {
+				if !errors.IsNotFound(err) {
+					return nil, fmt.Errorf("failed to get cluster infrastructure info: %v", err)
+				}
+			}
+			if cloudInfra != nil &&
+				cloudInfra.Status.PlatformStatus != nil &&
+				cloudInfra.Status.PlatformStatus.OpenStack != nil {
+				cloudName = cloudInfra.Status.PlatformStatus.OpenStack.CloudName
+			}
+			if len(cloudName) == 0 {
```

Comment:
Are you also trying to capture if cloudInfra.Status.PlatformStatus.OpenStack.CloudName is blank here?
Just curious.

Reply:
Yes, so we can unblock the install team. If we don't have the infra object, or the platform status doesn't have the cloud name filled yet, fall back to `openstack`.